x/crosschain/keeper/keeper_cross_chain_tx_prove_outbound_tx.go

```diff
@@ -0,0 +1,195 @@
+//	observation --> fail_old_status: Observation failed
+//	fail_old_status --> PendingRevert: Old status is PendingOutbound
+//	fail_old_status --> Aborted: Old status is PendingRevert
+//	PendingOutbound --> Aborted: Finalize outbound error
+//
+// ```
+//
+// Only observer validators are authorized to broadcast this message.
+func (k msgServer) ProveOutboundTx(goCtx context.Context, msg *types.MsgProveOutboundTx) (*types.MsgProveOutboundTxResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	// verify proof
+	blockHash := eth.HexToHash(msg.BlockHash)
+	res, found := k.zetaObserverKeeper.GetBlockHeader(ctx, blockHash.Bytes())
+	if !found {
+		return nil, sdkerrors.Wrapf(observerTypes.ErrBlockHeaderNotFound, "Block header not found: %s", msg.BlockHash)
+	}
+	chainID := res.ChainId
+
+	var header ethtypes.Header
+	err := rlp.DecodeBytes(res.Header, &header)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(observerTypes.ErrBlockHeaderNotFound, "cannot decode block header: %s", msg.BlockHash)
+	}
+	var txx ethtypes.Transaction
+	var receipt ethtypes.Receipt
+	txxVal, err := msg.TxProof.Verify(header.TxHash, int(msg.TxIndex))
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot verify tx proof: %s", err)
+	}
+	receiptVal, err := msg.ReceiptProof.Verify(header.ReceiptHash, int(msg.TxIndex))
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot verify receipt proof: %s", err)
+	}
+	err = txx.UnmarshalBinary(txxVal)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot unmarshal tx: %s", err)
+	}
+	err = receipt.UnmarshalBinary(receiptVal)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProofVerificationFail, "cannot unmarshal receipt: %s", err)
+	}
+	// end verify proof
+
+	tss, found := k.GetTSS(ctx)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrCannotFindTSSKeys, "Tss not found")
+	}
+	nonce := txx.Nonce()
+	cctxToNonce, found := k.GetNonceToCctx(ctx, tss.TssPubkey, chainID, int64(nonce))
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrCannotFindCctx, "Cctx not found")
+	}
+	index := cctxToNonce.CctxIndex
+	cctx, found := k.GetCrossChainTx(ctx, index)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrCannotFindCctx, "Cctx not found")
+	}
+	status := cctx.CctxStatus.Status
+	if status != types.CctxStatus_PendingOutbound && status != types.CctxStatus_PendingRevert {
+		return nil, sdkerrors.Wrapf(types.ErrStatusNotPending, "Cctx status is not pending")
+	}
+
+	// this is important; otherwise smoketest "Goerli->Goerli Message Passing (revert fail)"
+	// will appear to be successful but it's not correct, because the same outtx are used
+	// accepted here. The scond one should be rejected because the nonce is not correct
+	if cctx.GetCurrentOutTxParam().OutboundTxTssNonce != nonce {
+		return nil, sdkerrors.Wrapf(types.ErrNonceMismatch, "Nonce mismatch")
+	}
+
+	if receipt.Status == ethtypes.ReceiptStatusFailed {
+
+	}
```

Comment:
Seems to be missing logic here

Reply:
This is PoC, directly confirm outbound by proof. 
We would not enable this at first.